base/file.jl

```diff
@@ -436,8 +436,13 @@ function readlink(path::AbstractString)
     end
 end
 
-function chmod(p::AbstractString, mode::Integer)
-    err = ccall(:jl_fs_chmod, Int32, (Cstring, Cint), p, mode)
+function chmod(path::AbstractString, mode::Integer; recursive::Bool=false)
+    if recursive && isdir(path)
+        for p in readdir(path)
+            chmod(joinpath(path, p), mode, recursive=true)
+        end
+    end
+    err = ccall(:jl_fs_chmod, Int32, (Cstring, Cint), path, mode)
```

Comment:
I often need to do `chmod -R u+w` on directory trees where I don't have write permission. This indeed only works if the permission is set first.

However, what about the converse, `chmod -R u-w`?